File: Models/CommentModel.cs
namespace RSPOCourseWork.Models;

public class CommentModel
{
    public int Id { get; set; }
    public string Text { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public UserModel User { get; set; }
    public AnimeModel? Anime { get; set; }
}

File: Frontend/react-anime-universe/src/models/comment.model.ts
import { AnimeDto } from "./anime.models";
import { IUser } from "./user.model";

export class CommentModel {
    constructor(
        public id: number,
        public name: string,
        public text: string,
        public createdAt: string,
        public updatedAt: string,
        public user: IUser,
        public anime: AnimeDto,
    ) {}
}

export class SetCommentModel {
    constructor(
        public name: string,
        public text: string,
        public createdAt?: string,
        public updatedAt?: string,
    ) {}
}
